### ISSUE TITLE
implement the `auth` subcommand

### ISSUE THREAD

**Issue by ahl** [oxidecomputer/oxide.rs#1]
The code from https://github.com/oxidecomputer/cli/blob/main/src/cmd_auth.rs is present, but mostly commented out. We'll need to consider some fundamental options such as what our CLI-wide (i.e. global) state looks like, what the on-disk / env-var config looks like, etc.

- [x] `auth login` #7 
- [x] `auth logout` https://github.com/oxidecomputer/oxide-sdk-and-cli/pull/73
- [x] `auth status` https://github.com/oxidecomputer/oxide-sdk-and-cli/pull/48

**Comment by karencfv** [comment]
Some discussion around this topic was started in this [issue](https://github.com/oxidecomputer/oxide-sdk-rust/issues/39) :)

**Comment by ahl** [comment]
@karencfv I think #6 is also relevant. Can you summarize the conclusions of the issue you linked?

**Comment by ahl** [comment]
@karencfv can we close this out?

**Comment by karencfv** [comment]
Sounds good!
